src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -57,98 +59,283 @@ public Class<CreateCollectionCommand> getCommandClass() {
+   * the appropriateness of model dimensions. It ensures that all required and type-specific
+   * conditions are met for the service to be considered valid.
+   *
+   * @param userConfig The user input vectorize service configuration.
+   * @param userVectorDimension The dimension specified by the user, may be null.
+   * @return The dimension to be used for the vector, should be from the internal configuration. It
+   *     will be used for auto populate the vector dimension
+   * @throws JsonApiException If the service configuration is invalid or unsupported.
+   */
+  private Integer validateService(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
+      Integer userVectorDimension) {
+    // Only for internal tests
+    if (userConfig.provider().equals("custom")) {
+      return userVectorDimension;
+    }
+    // Check if the service provider exists and is enabled
+    PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig =
+        getAndValidateProviderConfig(userConfig);
+
+    // Check secret name for shared secret authentication, if applicable
+    validateAuthentication(userConfig, providerConfig);
+
+    // Validate user-provided parameters against internal expectations
+    validateUserParameters(userConfig, providerConfig);
+
+    // Validate the model and its vector dimension
+    return validateModelAndDimension(userConfig, providerConfig, userVectorDimension);
+  }
+
+  private PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig getAndValidateProviderConfig(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig) {
+    PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig =
+        embeddingProviderConfig.providers().get(userConfig.provider());
+    if (providerConfig == null || !providerConfig.enabled()) {
+      throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+          "Service provider '%s' is not supported", userConfig.provider());
+    }
+    return providerConfig;
+  }
+
+  // TODO: validate the 'secret_name' in the future
+  private void validateAuthentication(
+      CreateCollectionCommand.Options.VectorSearchConfig.VectorizeConfig userConfig,
+      PropertyBasedEmbeddingProviderConfig.EmbeddingProviderConfig providerConfig) {
+    // Check if user authentication type is support
+    userConfig.vectorizeServiceAuthentication().type().stream()
+        .filter(type -> !providerConfig.supportedAuthentication().contains(type))
+        .findFirst()
+        .ifPresent(
+            type -> {
+              throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
+                  "Authentication type '%s' is not supported", type);
+            });
+    // Check if 'secret_name' is provided if authentication type is 'SHARED_SECRET'
+    if (userConfig.vectorizeServiceAuthentication().type().contains("SHARED_SECRET")
```

Comment:
This should throw error now to state `SHARED_SECRET` type currently now supported.